solidity/contracts/BancorNetwork.sol

```diff
@@ -106,6 +107,20 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractRegistryClient {
         etherTokens[_token] = _register;
     }
 
+    /**
+      * @dev returns the conversion path between two tokens in the network
+      * note that this method is quite expensive in terms of gas and should generally be called off-chain
+      * 
+      * @param _sourceToken source token address
+      * @param _targetToken target token address
+      *
+      * @return conversion path between the two tokens
+    */
+    function conversionPath(IERC20Token _sourceToken, IERC20Token _targetToken) public view returns (address[]) {

```

Comment:
@yudilevi : This would cost more gas (when called from a transaction) and more provider computation time (when called from an RPC) than calling it directly from the path-finder.
The only reason in favor of this as far as I can think of, is that it will allow us to change the path-finder implementation without changing the network implementation (and so if you consider redeploying the network as an API-breaking move on our behalf, then this will allow us to avoid it).
Is that your primary goal here?

Reply:
yes, that's one reason and another reason is for users to know/worry about one less contract